encoder.go

```diff
@@ -384,14 +383,13 @@ func (e *Encoder) EncodeMapEntryHeader(tag int, size int) {
 }
 
 // stringToBytes is an optimized convert from a string to a []byte using unsafe.Pointer
-// nolint:gosec,govet,unsafeptr
 func (e *Encoder) stringToBytes(s string) []byte {
-	strh := (*reflect.StringHeader)(unsafe.Pointer(&s))
-	var sh reflect.SliceHeader
-	sh.Data = strh.Data
-	sh.Len = strh.Len
-	sh.Cap = strh.Len
-	return *(*[]byte)(unsafe.Pointer(&sh))
+	return *(*[]byte)(unsafe.Pointer(
```

Comment:
not sure if this should be part of this PR? otherwise looks good

Reply:
You have a point.  I included it because the change resolves a strange issue where this function was tripping the built-in linter in `gopls` and showing as a problem in VS Code.  The new logic does the same thing but doesn't trip up the editor.